model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -65,58 +65,50 @@ def find_and_replace_pattern(self, graph: Graph):
 
 
 class EltwiseInputReshape(MiddleReplacementPattern):
-    enabled = True
-    force_clean_up = True
-
-    def run_after(self):
-        from extensions.middle.pass_separator import MiddleStart
-        return [MiddleStart]
+    # This pass should be called directly from pipeline before layout change and other permutations
+    enabled = False
+    force_shape_inference = True
 
     def find_and_replace_pattern(self, graph: Graph):
-        for node in graph.get_data_nodes():
-            # Get all requested shapes for current node
-            # This mapping will contain pairs like {shape:[list of consumers nodes]}
-            mapping = {}
-            for consumer in node.out_nodes():
-                edge_attrs = graph.get_edge_data(node.id, consumer.id)[0]
-                if 'new_shape' in edge_attrs:
-                    if np.array_equal(edge_attrs['new_shape'], node.shape):
-                        continue
-                    new_shape = tuple([x for x in edge_attrs['new_shape']])
-                    if not new_shape in mapping:
-                        mapping.update({new_shape: [consumer]})
+        # Generate a map for producers of eltwise nodes with non-normalized shapes
+        # and in this map every producer has another map that reflects normalized shape
+        # to a list of eltwise-consumers
+        mapping = {}
+        for eltwise_node in graph.get_op_nodes(is_eltwise=True):
+            eltwise_shape = eltwise_node.out_port(0).data.get_shape()
+            for in_port_idx in eltwise_node.in_ports():
+                consumer_port = eltwise_node.in_port(in_port_idx)
+                producer_port = consumer_port.get_source()
+                producer_shape = producer_port.data.get_shape()
+                if len(producer_shape) == len(eltwise_shape):
+                    continue
+                unsqueeze_dims = []
+                if len(producer_shape) != len(eltwise_shape):
+                    for x in range(len(eltwise_shape) - len(producer_shape)):
+                        unsqueeze_dims.append(x)
+                unsqueeze_dims = tuple([x for x in unsqueeze_dims])
+                if not producer_port in mapping:
+                    mapping.update({producer_port: {unsqueeze_dims: [consumer_port]}})
+                else:
+                    int_mapping = mapping[producer_port].copy()
+                    if not unsqueeze_dims in int_mapping:
+                        int_mapping.update({unsqueeze_dims: [consumer_port]})
```

Comment:
Move this code undef `if` statement when `len(producer_shape) != len(eltwise_shape)`

Reply:
Done.